test/cli/test_gen.py

```diff
@@ -1,46 +1,55 @@
 import unittest
+from typing import List
+from abc import ABCMeta
+import click
+import click.testing
 
-from xcube.cli.gen import main, GenL2CHelpFormatter
+from xcube.cli.gen import generate_cube as cli
+# TODO AliceBalfanz: extended help section, reason for keeping code for GenL2CHelpFormatter
+# from xcube.cli.gen import main, GenL2CHelpFormatter
 
 
-class CliTest(unittest.TestCase):
-    def test_main_with_option_help(self):
-        exit_code = main(['-h'])
-        self.assertEqual(0, exit_code)
-        exit_code = main(['--help'])
-        self.assertEqual(0, exit_code)
+class CliTest(unittest.TestCase, metaclass=ABCMeta):
 
-    def test_main_with_missing_args(self):
-        exit_code = main([])
+    @classmethod
+    def invoke_cli(cls, args: List[str]):
+        runner = click.testing.CliRunner()
+        return runner.invoke(cli, args)
+
+    def test_help_option(self):
+        result = self.invoke_cli(['time_irregular', '--help'])
+        self.assertEqual(0, result.exit_code)
+
+    def test_missing_args(self):
+        result = self.invoke_cli([])
         # exit_code==0 when run from PyCharm, exit_code==2 else
-        self.assertEqual(2, exit_code)
-        #self.assertTrue(exit_code in [0, 2])
+        self.assertEqual(1, result.exit_code)
 
     def test_main_with_illegal_size_option(self):
-        exit_code = main(['-s', '120,abc', 'input.nc'])
```

Comment:
For me, the name `exit_code` was clearer because it also reflected the role of the return value. In fact, the CLI will use this value as exit code in the shell.